i386/Dockerfile

```diff
@@ -2,9 +2,22 @@ FROM ioft/i386-ubuntu:16.04
 MAINTAINER Jonne Haß <me@jhass.eu>
 
 RUN apt-get update && \
-apt-get install -y apt-transport-https curl build-essential pkg-config libssl-dev llvm libedit-dev libgmp-dev libxml2-dev libyaml-dev libreadline-dev git-core && \
-curl https://dist.crystal-lang.org/apt/setup.sh | bash && \
+apt-get install -y apt-transport-https curl build-essential pkg-config libssl-dev llvm libedit-dev libgmp-dev libxml2-dev libyaml-dev libreadline-dev git-core binutils-dev libcurl4-openssl-dev zlib1g-dev libdw-dev libiberty-dev cmake python
+
+RUN curl https://dist.crystal-lang.org/apt/setup.sh | bash && \
 apt-get update && \
-apt-get install -y crystal
+apt-get install -y crystal && \
+\
+git clone https://github.com/SimonKagstrom/kcov /tmp/kcov && \
```

Comment:
Since we're only checking out a tag and don't need the repository's entire history, we could instead download and extract a repository tarball:

```sh
mkdir /tmp/kcov && cd /tmp/kcov && \
  curl -fSL https://github.com/SimonKagstrom/kcov/archive/v33.tar.gz | \
  tar -xzf - --strip-components=1
```